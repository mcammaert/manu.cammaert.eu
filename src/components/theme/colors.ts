export const colors = {
  primary90: '#0c2d48',
  primary80: '#0b395d',
  primary70: '#0b4573',
  primary60: '#0d5189',
  primary50: '#145da0',
  primary40: '#367bb1',
  primary30: '#5c99c1',
  primary20: '#85b7d0',
  primary10: '#b1d4e0',

  secondary90: '#ca3f36',
  secondary80: '#d35239',
  secondary70: '#db633e',
  secondary60: '#e37443',
  secondary50: '#eb844a',
  secondary40: '#f79b77',
  secondary30: '#fdb5a1',
  secondary20: '#fdd0c8',
  secondary10: '#f7eceb',

  accent1: '#d45e95',
  accent2: '#b290d4',
  accent3: '#97b8e6',

  // 10 shades of grey
  black: '#1E2025',
  charcoal: '#262830',
  shadow: '#2E313A',
  lead: '#585965',
  steel: '#787C8A',
  gray: '#989DAC',
  smoke: '#BEC2C9',
  fog: '#EDEEF0',
  white: '#F6F7F9',
  snow: '#FCFCFC',

  // slate
  slateLight: '#3279AD',
  slate: '#4C6E89',
  slateDark: '#455C77',

  // teal
  teal: '#3D829C',
  tealDark: '#36768B',

  // laguna
  lagunaLight: '#4FB8F3',
  laguna: '#57A7E3',
  lagunaDark: '#3680DC',

  // warm colors
  red: '#F60C5B',
  orange: '#F86C5C',
  sunflower: '#EDC172',
  lemon: '#FFE362',

  // purples
  grape: '#374278',
  plum: '#725FA4',
  lavender: '#936FBA',
  purple: '#C163C6',

  // greens
  forest: '#50C04E',
  lawn: '#86DE6A',
  aquamarine: '#14DAB5',
  emerald: '#0DCACE',
  turquoise: '#57D5D6',

  // browns
  chocolate: '#3C322D',
  barnwood: '#5E5E5F',
};

export default colors;
