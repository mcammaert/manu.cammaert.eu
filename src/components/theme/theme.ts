import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  body {
    background-color: #ccc;
  }
`;

const theme = {
  main: 'mediumseagreen',
};

export default theme;
