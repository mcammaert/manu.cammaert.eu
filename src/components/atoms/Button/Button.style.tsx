import styled, { css } from 'styled-components';
import { rgba, modularScale } from 'polished';
import { colors } from 'components/theme/colors';

export const button = css`
  text-decoration: none;
  font-size: ${modularScale(0)};
  font-weight: 400;
  padding: ${modularScale(-2)} ${modularScale(-1)};
  color: ${({ theme }) => theme.colors.secondaryInvert};
  background: ${({ theme }) => theme.colors.secondary};
  cursor: pointer;
  border: none;
  border-radius: 6px;
  box-shadow: 0 1px 18px ${rgba(colors.black, 0.08)};
  opacity: 0.8;
  transition: all 200ms;
  &:hover:enabled {
    opacity: 1;
    cursor: pointer;
    box-shadow: 0 1px 18px ${rgba(colors.black, 0.25)};
  }
  &:active {
    color: ${({ theme }) => theme.colors.secondary};
    background: ${({ theme }) => theme.colors.secondaryInvert};
  }
  &:disabled {
    opacity: 0.75;
    cursor: default;
    color: ${({ theme }) => theme.colors.secondaryInvert};
    background: ${({ theme }) => theme.colors.secondary};
  }
  -webkit-appearance: initial;
`;

export const Button = styled.button.attrs({ type: 'button' })`
  ${button}
`;

export default { Button };
