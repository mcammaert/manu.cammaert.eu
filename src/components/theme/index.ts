import { createGlobalStyle } from 'styled-components';
import 'normalize.css';
import { colors } from './colors';

export interface GlobalStyleProps {
  readonly light: boolean;
}

// const backgroundColor = (location: string) => (location === '/' ? colors.oxfordBlue : colors.fuzzyWuzzy);

export const GlobalStyle = createGlobalStyle<GlobalStyleProps>`
  body {
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    font-family: 'Nunito Sans', sans-serif;
    width: 100vw;
    background: ${({ light }) => (light ? colors.white : colors.black)};
    background-attachment: fixed;    
    color: ${({ light }) => (light ? colors.black : colors.snow)};
    transition: all 800ms;
    a {
      color: ${({ theme }) => theme.colors.secondary};
    }
  }
`;

export const lightTheme = {
  margin: 15,
  colors: {
    primary: colors.charcoal,
    primaryInvert: colors.snow,
    secondary: colors.primary60,
    secondaryInvert: colors.white,
    tertiary: colors.secondary80,
    tertiaryInvert: colors.white,
    copy: colors.charcoal,
    copyInvert: colors.white,
  },
};

export const darkTheme = {
  margin: 15,
  colors: {
    primary: colors.snow,
    primaryInvert: colors.black,
    secondary: colors.primary30,
    secondaryInvert: colors.primary90,
    tertiary: colors.secondary30,
    tertiaryInvert: colors.secondary80,
    copy: colors.white,
    copyInvert: colors.charcoal,
  },
};
