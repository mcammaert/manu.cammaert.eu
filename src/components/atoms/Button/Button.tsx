import React from 'react';

import { ButtonProps } from './Button.types';

import S from './Button.style';

const Button: React.FC<ButtonProps> = ({ children, onClick, disabled = false }) => (
  <S.Button onClick={onClick} disabled={disabled}>
    {children}
  </S.Button>
);

export default Button;
