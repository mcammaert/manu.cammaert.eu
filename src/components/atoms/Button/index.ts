import * as S from './Button.style';

export { default as Button } from './Button';
export * from './Button.types';
export { S };
