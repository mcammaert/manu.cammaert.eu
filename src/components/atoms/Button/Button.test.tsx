import React from 'react';

import { cleanup } from '@testing-library/react';
import 'jest-styled-components';

import { renderComponent } from 'testing/react-testing-library';
import { lightTheme } from 'components/theme';

import Button from './Button';

afterEach(cleanup);

describe('Button', () => {
  describe('default props', () => {
    it('should match the snapshot', () => {
      const component = <Button onClick={() => {}} />;
      const { container } = renderComponent({ theme: lightTheme, component });
      expect(container).toMatchSnapshot();
    });
  });
});
