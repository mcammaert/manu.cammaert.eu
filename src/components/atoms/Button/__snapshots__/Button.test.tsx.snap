// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button default props should match the snapshot 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1em;
  font-weight: 400;
  padding: 0.5627813555039173em 0.7501875468867217em;
  color: #F6F7F9;
  background: #0d5189;
  cursor: pointer;
  border: none;
  border-radius: 6px;
  box-shadow: 0 1px 18px rgba(30,32,37,0.08);
  opacity: 0.8;
  -webkit-transition: all 200ms;
  transition: all 200ms;
  -webkit-appearance: initial;
}

.c0:hover:enabled {
  opacity: 1;
  cursor: pointer;
  box-shadow: 0 1px 18px rgba(30,32,37,0.25);
}

.c0:active {
  color: #0d5189;
  background: #F6F7F9;
}

.c0:disabled {
  opacity: 0.75;
  cursor: default;
  color: #F6F7F9;
  background: #0d5189;
}

<div>
  <button
    class="c0"
    type="button"
  />
</div>
`;
